notebooks/search/07-inference.ipynb

```diff
@@ -0,0 +1,382 @@
+    "    # For local development\n",
+    "    # hosts=[\"http://localhost:9200\"] \n",
+    "    cloud_id=ELASTIC_CLOUD_ID,\n",
+    "    api_key=ELASTIC_API_KEY,\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "b1115ffb",
+   "metadata": {},
+   "source": [
+    "Confirm that the client has connected with this test:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "cc0de5ea",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "print(client.info())"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "4e9e7354",
+   "metadata": {},
+   "source": [
+    "Refer to [the documentation](https://www.elastic.co/guide/en/elasticsearch/client/python-api/current/connecting.html#connect-self-managed-new) to learn how to connect to a self-managed deployment.\n",
+    "\n",
+    "Read [this page](https://www.elastic.co/guide/en/elasticsearch/client/python-api/current/connecting.html#connect-self-managed-new) to learn how to connect using API keys.\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "96788aa1",
+   "metadata": {},
+   "source": [
+    "## Create the inference task\n",
+    "\n",
+    "Let's create the inference task by using the [Create inference API](https://www.elastic.co/guide/en/elasticsearch/reference/current/put-inference-api.html).\n",
+    "\n",
+    "You'll need an OpenAI API key for this that you can find in your OpenAI account under the [API keys section](https://platform.openai.com/api-keys). A paid membership is required to complete the steps in this notebook as the OpenAI free trial API usage is limited."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "3e6d98af",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "API_KEY = getpass.getpass('Enter OpenAI API key:  ')\n",
+    "\n",
+    "client.inference.put_model(\n",
+    "    task_type=\"text_embedding\",\n",
+    "    model_id=\"openai_embeddings\",\n",
+    "    body={\n",
+    "        \"service\": \"openai\",\n",
+    "        \"service_settings\": {\n",
+    "            \"api_key\": API_KEY\n",
+    "        },\n",
+    "        \"task_settings\": {\n",
+    "            \"model\": \"text-embedding-ada-002\"\n",
+    "        }\n",
+    "    }\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "e5feaf12",
+   "metadata": {},
+   "source": [
+    "## Create an ingest pipeline with an inference processor\n",
+    "\n",
+    "Create an ingest pipeline with an inference processor by using the [`put_pipeline`](https://www.elastic.co/guide/en/elasticsearch/reference/master/put-pipeline-api.html) method. Reference the OpenAI model created above to infer against the data that is being ingested in the pipeline."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "c5897fe4",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "client.ingest.put_pipeline(\n",
+    "    id=\"openai_embeddings\", \n",
```

Comment:
Might be confusing that you are using the same `openai_embeddings` name for `model_id` and for the pipeline `id`. I would either add a note indicating why, or maybe better to just use different names.